### ISSUE TITLE
Subject + buffer() prevents flow cancellation

### ISSUE THREAD

**Issue by guelo** [akarnokd/kotlin-flow-extensions#29]
Something about adding `subject.buffer()` prevents collection cancellation. In this test I use `take(1)` to cancel the flow.

```kotlin
@Test(timeout=1000) // wait a second 
fun test() = runBlocking {

    val subject = PublishSubject<Int>()

    val job = launch {
        subject
            .buffer()
            .take(1) // cancel after first emission
            .collect { println("$it") }
        println("Done")
    }

    // wait for the collector to arrive
    while (!subject.hasCollectors()) {
        delay(1)
    }

    subject.emit(1)

    job.join()
}
```
"Done" is never printed and the junit timeout is triggered. If I make a second call to emit() it does complete without printing the second emission. The same code using `Channel<Int>().consumeAsFlow().buffer().take(1)` works properly. 

Is there some kind of cooperative cancellation that's missing in the Subjects?

**Comment by akarnokd** [comment]
Probably. I'll look into it.

**Comment by akarnokd** [comment]
Fixed via v0.0.4
